charts/internal/calico/templates/multus/daemonset-multus-cni.yaml

```diff
@@ -0,0 +1,136 @@
+{{- if .Values.config.multus.enabled }}
+---
+# This manifest installs Multus CNI plugins and network config on
+# each node in a Kubernetes cluster.
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: multus
+  namespace: kube-system
+  labels:
+    k8s-app: multus
+    gardener.cloud/role: system-component
+    node.gardener.cloud/critical-component: "true"
+  {{- if not .Values.autoscaling.staticRequests }}
+  annotations:
+    resources.gardener.cloud/preserve-resources: 'true'
+  {{- end }}
```

Comment:
I think we do not need this unless we autoscale the daemonset, which does not seem to be the case right now.

What does this daemonset actually do at runtime besides installing the CNI binaries with the init container?

Reply:
I think I have copied this bit over from `calico-node` DaemonSet.

Agree, it's not doing much. Though the running the main container with `thin_entrypoint` is also important that it enters the watch (based on conditions https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/cmd/thin_entrypoint/main.go#L621-L634 ) and updates the multus kubeconfig:

```shell
› kubectl-v1.31.5 -n kube-system logs multus-g4gzj
Defaulted container "multus" out of: multus, install-multus-cni (init), install-cni-plugins (init)
CA (false) or SA token (true) changed - recreating kubeconfig
kubeconfig is created in /host/etc/cni/net.d/multus.d/multus.kubeconfig
```